test/node-fetch/main.js

```diff
@@ -63,111 +49,110 @@ describe('node-fetch', () => {
   it('should return a promise', () => {
     const url = `${base}hello`
     const p = fetch(url)
-    expect(p).to.be.an.instanceof(Promise)
-    expect(p).to.have.property('then')
+    assert.ok(p instanceof Promise)
+    assert.strictEqual(typeof p.then, 'function')
   })
 
   it('should expose Headers, Response and Request constructors', () => {
-    expect(Headers).to.equal(HeadersOrig)
-    expect(Response).to.equal(ResponseOrig)
-    expect(Request).to.equal(RequestOrig)
+    assert.strictEqual(Headers, HeadersOrig)
+    assert.strictEqual(Response, ResponseOrig)
+    assert.strictEqual(Request, RequestOrig)
   })
 
   it('should support proper toString output for Headers, Response and Request objects', () => {
-    expect(new Headers().toString()).to.equal('[object Headers]')
-    expect(new Response().toString()).to.equal('[object Response]')
-    expect(new Request(base).toString()).to.equal('[object Request]')
+    assert.strictEqual(new Headers().toString(), '[object Headers]')
+    assert.strictEqual(new Response().toString(), '[object Response]')
+    assert.strictEqual(new Request(base).toString(), '[object Request]')
   })
-
+  // TODO Should we reflect the input?
   it('should reject with error if url is protocol relative', () => {
     const url = '//example.com/'
-    return expect(fetch(url)).to.eventually.be.rejectedWith(TypeError)
+    return assert.rejects(fetch(url), new TypeError('Failed to parse URL from //example.com/'))
   })
 
   it('should reject with error if url is relative path', () => {
     const url = '/some/path'
-    return expect(fetch(url)).to.eventually.be.rejectedWith(TypeError)
+    return assert.rejects(fetch(url), new TypeError('Failed to parse URL from /some/path'))
   })
 
+  // TODO: This seems odd
```

Comment:
Can you explain why this seems odd?

I am seeing the following:

```
TypeError: fetch failed
    at node:internal/deps/undici/undici:12502:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  [cause]: Error: unknown scheme
```

Perhaps this test should check for `unknown scheme` instead?

Reply:
Yeah, I think this is what made it odd to me.

Reply:
fetch should *always* reject with "fetch failed" and the cause should be the error. Basically, it's working properly.

Reply:
I see. In that case we should remove the comment?

Reply:
I think are lacking the test that error.cause.message should be "unknown scheme".